### PR TITLE
Teach tab to focus terminal after rename

### PR DESCRIPTION
**Pull request by Don-Vito**
## Summary of the Pull Request
After rename ends (either by enter or escape) the rename box
gets collapsed and the focus moves to the next tab stop
in the TabView (e.g., new tab button).

To overcome this:
* Added RenameEnded event to TabHeaderControl
* Forwarded it as RenamerDeactivated from TerminalTab
* Registered in the TerminalPage to focus the active control upon
RenamerDeactivated if focus didn't move to tab menu.

This means, no matter how you close the renamer,
the current terminal gains the focus.

## PR Checklist
* [x] Closes https://github.com/microsoft/terminal/issues/9160
* [x] CLA signed. 
* [ ] Tests added/passed
* [ ] Documentation updated. 
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already.

### CODE REVIEW

**Review by carlos-zamora — APPROVED** [review]
_Love it!_ Awesome work!

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -858,6 +858,22 @@ namespace winrt::TerminalApp::implementation
...
+            if (const auto page{ weakThis.get() })
+            {
+                if (!page->_newTabButton.Flyout().IsOpen())
+                {
+                    if (const auto tab{ page->_GetFocusedTabImpl() })
+                    {
+                        if (const auto control{ tab->GetActiveTerminalControl() })
+                        {
+                            control.Focus(FocusState::Programmatic);
+                        }
```
Comment (PankajBhojwani) [inline src/cascadia/TerminalApp/TerminalPage.cpp:871]: Nit: should we just do `tab->Focus(FocusState::Programmatic)` to keep it consistent with the rest of the file?

Comment (Don-Vito) [reply src/cascadia/TerminalApp/TerminalPage.cpp:871]: Love it! Thanks!

**Review by zadjii-msft — APPROVED** [review]
You know, we've got a mess of focus issues, but I don't think this makes it worse. I think I was hoping that there was some sort of holistic "whenever the tab gets focus, for any reason, toss the focus back to the content" solution to our focus issues. This is more of a specific solution to an individual case of focus being weird. It might just be easier to solve these issues on-off as they come up.

In the future, we might want to pull the `TabRenamerDeactivated` lambda into it's own method in TerminalPage, so that other scenarios might be able to leverage this as well.

**Comment by Don-Vito** [comment]
I think the CI is stuck. Can someone kick it? :blush:

**Pull request merged**